cognite/client/_api/datapoints.py

```diff
@@ -756,7 +752,12 @@ def _insert_datapoints_concurrently(self, dps_object_lists: List[List[Dict[str,
         )
 
     def _insert_datapoints(self, post_dps_objects: List[Dict[str, Any]]):
+        # convert to memory intensive format as late as possible and clean up after
+        for it in post_dps_objects:
+            it["datapoints"] = [{"timestamp": t, "value": v} for t, v in it["datapoints"]]
         self.client._post(url_path=self.client._RESOURCE_PATH, json={"items": post_dps_objects})
+        for it in post_dps_objects:
+            del it["datapoints"]
```

Comment:
Does this make any difference? Won't the garbage collector take care of these as soon as the method returns and references to post_dbs_objects are dropped?

Reply:
I thought there is still a reference to it via the `tasks` object whose elements are being modified, and these won't be cleared up until _insert_datapoints_concurrently returns (and all of these converted objects pile up). No?

Reply:
Right. That's probably correct